test/wrappers/ortools_test.rb

```diff
@@ -5770,4 +5770,31 @@ def test_self_selection_should_not_change_vehicle_ids
       end
     end
   end
+
+  def test_quantity_precision
+    problem = VRP.lat_lon_two_vehicles
+    problem[:services].each{ |service|
+      service[:quantities] = [{ unit_id: 'kg', value: 1.001 }]
+    }
+    problem[:vehicles].each{ |vehicle|
+      vehicle[:capacities] = [{ unit_id: 'kg', limit: 13 }]
+    }
+
+    result = OptimizerWrapper.wrapper_vrp('demo', { services: { vrp: [:ortools] }}, TestHelper.create(problem), nil)
+    assert_equal 13, result[:routes].map{ |route| route[:activities].count{ |activity| activity[:service_id] } }.sum,
+                 'The problem and result are expected to contain 13 services'
+
```

Comment:
We don't care about this in this test, if (for any reason, like performance of the machine etc) or-tools or vroom returns a single woute with 12 services planned and one unplanned, it is not violating the `test_quantity_precision` so the test should pass. We have proper performance tests for this.

Reply:
https://github.com/Mapotempo/optimizer-api/pull/213#discussion_r626393005

As asked by @fonsecadeline whenever we change the instance, the test will become obsolete and nothing will break

Reply:
Right, I was suggesting something more like 

```suggestion
    assert_equal 13, result[:routes].map{ |route| route[:activities].count{ |activity| activity[:service_id] } }.sum + result[:unassigned].size,
                 'The problem and result are expected to contain 13 services'

```

You can eventually add something like 

```
assert result[:routes].any?{ |r| r[:activities].any{ |a| a[:service_id] } }
```

to make sure we were able to assign services but we did not overpass limit.

Reply:
Maybe we can just but a limit of 4 and avoid all these checks (because I do not think we would reduce instance with less than 4 services).

Reply:
I didn't think of the case of empty solution, so yeah we can have something like that (`assert result[:routes].any?{ |r| r[:activities].any?{ |a| a[:service_id] } }, 'All services are unplanned'`)

The first one doesn't work if everything is unassigned... and we have that exact check in the wrapper_vrp for everything we optimise.

(We should never change the instances exactly because of this reason, these are "fixed" instances, except for obvious/simple correction, we never touch the VRP.instance type instances)

Reply:
This (no change) is the rule on the mt-web side for their fixed instances as well; otherwise, you lose the validity of all the tests using the instance.